src/lib.rs

```diff
@@ -47,31 +53,25 @@ pub trait Apps<C: trussed::Client, D> {
     ) -> T;
 }
 
-pub struct Runner<S: StoreProvider> {
+pub trait ClientBuilder<C, D: Dispatch> {
+    fn build(&self, id: &str, backends: &'static [BackendId<D::BackendId>]) -> C;
+}
```

Comment:
Generally, I try to avoid that (unless the trait is already non-object-safe) because it would make the trait not object safe.  In this specific case, the implementer also has to fix `D`, see for example:

```rust
impl<S: StoreProvider, D: Dispatch> ClientBuilder<Client<S, D>, D> for Service<S, D> { ... }
```

`Service` can only build clients with its specific dispatch, not any dispatch imlpementation.  So I don’t think it would work here.